tests/ConversationTranslatorTests.ts

```diff
@@ -84,6 +86,401 @@ describe("conversation constructor tests", () => {
+                    ((error: any) => {
+                        done.fail();
+                    }));
+            });
+        }),
+        ((error: any) => {
+            done.fail();
+        }));
+        objsToClose.push(c);
+
+        function joinParticipant(code: string): void {
+            // join as a participant
+            const ctP: sdk.ConversationTranslator = new sdk.ConversationTranslator();
+            if (endpointHost !== "") { ctP.properties.setProperty(sdk.PropertyId.ConversationTranslator_Host, endpointHost); }
+            if (speechEndpointHost !== "") { ctP.properties.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+            objsToClose.push(ctP);
+            ctP.joinConversationAsync(code, "mute me", "en-US",
+                    (() => {
+                        // continue
+                    }),
+                    ((error: any) => {
+                        done.fail();
+                    }));
+        }
+
+        WaitForCondition(() => (commandCount > 2), done);
+
+    }, 20000);
+
+    test("Start Conversation, join as host and send message", (done: jest.DoneCallback) => {
+
+        // tslint:disable-next-line:no-console
+        console.info("Start Conversation, join as host and send message");
+
+        let textMessage: string = "";
+
+        // start a conversation
+        const config = sdk.SpeechTranslationConfig.fromSubscription(Settings.SpeechSubscriptionKey, Settings.SpeechRegion);
+        if (endpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.ConversationTranslator_Host], endpointHost); }
+        if (speechEndpointHost !== "") { config.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+        const c: sdk.Conversation = sdk.Conversation.createConversationAsync(config, (() => {
+            objsToClose.push(c);
+
+            const ct: sdk.ConversationTranslator = new sdk.ConversationTranslator();
+            if (endpointHost !== "") { ct.properties.setProperty(sdk.PropertyId.ConversationTranslator_Host, endpointHost); }
+            if (speechEndpointHost !== "") { ct.properties.setProperty(sdk.PropertyId[sdk.PropertyId.SpeechServiceConnection_Host], speechEndpointHost); }
+
+            ct.participantsChanged = ((s: sdk.ConversationTranslator, e: sdk.ConversationParticipantsChangedEventArgs) => {
+                sendMessage(`Hello ${e.participants[0].displayName}`);
+            });
+            ct.textMessageReceived = ((s: sdk.ConversationTranslator, e: sdk.ConversationTranslationEventArgs) => {
+                // tslint:disable-next-line: no-console
+                console.log(`text message received: ${e.result.text}`);
+                textMessage = e.result.text;
+            });
+
+            function sendMessage(message: string): void {
+                ct.sendTextMessageAsync(message);
+            }
+
+            objsToClose.push(ct);
```

Comment:
> objsToClose.push(ct); [](start = 11, length = 22)

This should be pushed onto the stack near creation.

Reply:
Ct objects updated